smdebug/tensorflow/collection.py

```diff
@@ -148,9 +148,6 @@ def __init__(self, collections=None, create_default=True):
                 self.create_collection(n)
             if is_tf_version_2x() and tf.executing_eagerly():
                 self.get(CollectionKeys.BIASES).include("^(?!gradient).*bias")
-                self.get(CollectionKeys.WEIGHTS).include("^weights/.*/((?!bias).)*$")
-                self.get(CollectionKeys.LOSSES).include(".*loss.*")
-                self.get(CollectionKeys.GRADIENTS).include("^gradient")
             else:
                 self.get(CollectionKeys.BIASES).include("bias")
```

Comment:
why weren't loss and othernregexes set for SessionHook ?

Reply:
Because SessionHook doesn't identify these tensors based on regex patterns. They come from global collections maintained by TF. (Not gradients where we manually add them)

Reply:
Why are the above being removed btw?

Reply:
If I have loss regex here, then https://github.com/awslabs/sagemaker-debugger/blob/master/smdebug/tensorflow/session.py#L206 starts adding other operations in the graph with the regex "*loss*" into the loss collection. It gave 105 loss tensors in the trial instead of 1.

since these regex patterns were added for gradient tape support, moved them to grad tape related function so that it doesn't affect other hooks